src/app/components/layouts/overlay/terra-overlay.component.spec.ts

```diff
@@ -1,11 +1,298 @@
+    });
+
+    it('should be initialized correctly', () =>
+    {
+        expect(component.inputIsSmall).toBe(false);
+        expect(component.inputIsLarge).toBe(true);
+        expect(component.inputIsExtraLarge).toBe(false);
+        expect(component.inputIsCloseable).toBe(true);
+        expect(component.inputIsStatic).toBe(false);
+
+        expect(component.inputOverlayTitle).toBeFalsy(); // default is not set
+
+        let header:DebugElement = fixture.debugElement.query(By.css('div.modal-header'));
+
+        expect(header).toBeTruthy(); // header should be present as default
+
+        expect(component.inputPrimaryButtonInterface).toBeFalsy();
+        expect(component.inputSecondaryButtonInterface).toBeFalsy();
+
+        let footer:DebugElement = fixture.debugElement.query(By.css('div.modal-footer'));
+
+        expect(footer).toBeFalsy(); // footer should NOT be present as default
+    });
+
+    it(`should call 'emitOutputOnHide' after it has been hidden`, () =>
+    {
+        let onOutputSpy:Spy = spyOn(component, 'emitOutputOnHide');
+        divElement.triggerEventHandler('onHidden', null);
+
+        expect(onOutputSpy).toHaveBeenCalled();
+    });
+
+    it(`should call 'emitOutputOnShow' after it has been hidden`, () =>
+    {
+        let onOutputSpy:Spy = spyOn(component, 'emitOutputOnShow');
+        divElement.triggerEventHandler('onShown', null);
+
+        expect(onOutputSpy).toHaveBeenCalled();
+    });
+
+    it(`should set modal sizes (classes) according to the inputs`, () =>
+    {
+        expect(modalDialogElement.classes['modal-lg']).toBe(true); // default
+
+        component.inputIsExtraLarge = true;
+        component.inputIsLarge = false;
+        component.inputIsSmall = true;
+
+        fixture.detectChanges();
+
+        expect(modalDialogElement.classes['modal-lg']).toBe(false);
+        expect(modalDialogElement.classes['modal-xl']).toBe(true);
+        expect(modalDialogElement.classes['modal-sm']).toBe(true);
+    });
+
+    it(`should set 'inputIsStatic' depending on 'inputIsClosable'`, () =>
+    {
+        component.inputIsCloseable = false;
+        fixture.detectChanges();
+        component.ngAfterViewInit();
+
+        expect(component.inputIsStatic).toBe(true);
+        expect(component.viewChildOverlay.config.backdrop).toBe('static');
+        expect(component.viewChildOverlay.config.keyboard).toBe(false);
+    });
+
+    it(`should show header depending on 'inputOverlayTitle'`, () =>
+    {
+        component.inputOverlayTitle = overlayTitle;
+        component.inputIsCloseable = false;
+
+        fixture.detectChanges();
+        let header:DebugElement = fixture.debugElement.query(By.css('div.modal-header'));
+
+        expect(header).toBeTruthy(); // should be present because inputOverlayTitle is set
+
+    });
+
+    it(`should show header depending on 'inputOverlayTitle' and 'inputIsClosable'`, () =>
+    {
+        component.inputOverlayTitle = overlayTitle;
```

Comment:
hier ggf nur auf inputIsClosable testen, sprich 
```suggestion
        component.inputOverlayTitle = null;
```

Reply:
ist schon in dem test drüber

Reply:
Naja eben nicht :-P